manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -131,6 +127,52 @@ public void deleteProcessor(String bridgeId, String processorId, String customer
                 processor.getBridge().getId());
     }
 
+    @Override
+    public ProcessorDTO toDTO(Processor processor) {
+        BridgeDTO bridgeDTO = processor.getBridge() != null ? bridgesService.toDTO(processor.getBridge()) : null;
+        ProcessorDefinition definition = processor.getDefinition() != null ? jsonNodeToDefinition(processor.getDefinition()) : null;
+        return new ProcessorDTO(processor.getId(), processor.getName(), definition, bridgeDTO, processor.getStatus());
+    }
+
+    @Override
+    public ProcessorResponse toResponse(Processor processor) {
+        ProcessorResponse processorResponse = new ProcessorResponse();
+
+        processorResponse.setId(processor.getId());
+        processorResponse.setName(processor.getName());
+        processorResponse.setStatus(processor.getStatus());
+        processorResponse.setPublishedAt(processor.getPublishedAt());
+        processorResponse.setSubmittedAt(processor.getSubmittedAt());
+
+        if (processor.getDefinition() != null) {
+            ProcessorDefinition definition = jsonNodeToDefinition(processor.getDefinition());
+            processorResponse.setFilters(definition.getFilters());
+            processorResponse.setTransformationTemplate(definition.getTransformationTemplate());
+            processorResponse.setAction(definition.getAction());
+        }
+
+        if (processor.getBridge() != null) {
+            processorResponse.setHref(APIConstants.USER_API_BASE_PATH + processor.getBridge().getId() + "/processors/" + processor.getId());
+            processorResponse.setBridge(bridgesService.toResponse(processor.getBridge()));
+        }
+
+        return processorResponse;
+    }
+
+    @Override
+    public JsonNode definitionToJsonNode(ProcessorDefinition definition) {
```

Comment:
@kostola This and the `jsonNodeToDefinition` methods are inheriting the `@Transactional` class-level annotation. They don't need a Transaction and given how frequently they are going to be called, I think we should remove it. Can you please remove the class level `@Transactional` annotation and move it only to the methods where a transaction is required.

Reply:
Done :+1: